packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -776,6 +799,55 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         return iconClass.concat(additionalClasses).join(' ');
     }
 
+    /**
+     * Returns the classname of the indent div based on the inputs
+     * @param needsNodeActiveGuideline "true" if active guideline is needed for either all siblings of a selected node or all child nodes of a selected parent node.
+     * @param needsLeafPadding "true" if we have reached the last indent of a child node.
+     */
+    protected renderIndentClass(needsNodeActiveGuideline: boolean, needsLeafPadding: boolean): string {
+        const indentGuideOption = this.corePreferences['workbench.tree.renderIndentGuides'];
+        const indentClassName = `${TREE_NODE_INDENT_BLOCK_CLASS}
+        ${indentGuideOption !== 'none' ? indentGuideOption === 'onHover' ? INDENT_GUIDE_ONHOVER : INDENT_GUIDE_ALWAYS : ''}  
+        ${needsLeafPadding ? TREE_NODE_CHILD_PADDING_CLASS : ''}
+        ${indentGuideOption !== 'none' && needsNodeActiveGuideline ? TREE_NODE_ACTIVE : ''}`;
+        return indentClassName;
+    }
+
+    /**
+     * Render indent for the file tree based on the depth
+     * @param node the tree node.
+     * @param depth the depth of the tree node.
+     */
+    protected renderIndent(node: TreeNode, props: NodeProps): React.ReactNode {
+        const indentDivs: React.ReactNode[] = [];
+        let nodePtr = node;
+        for (let i = 0; i < props.depth; i++) {
```

Comment:
this code looks rather complicated, could please formulate what are you trying to achieve here?

Reply:
@akosyakov The number of indents being rendered depend on the depth of the tree node. If a tree node has a depth of 3, we would render 3 indents for the node. 

To determine if we need to render an active guideline for an indent of a tree node, we look for the parent of the node that is currently being rendered. There are 2 cases which we render an active guideline for an indent, 1/ if the parent node is selected and 2/ if a sibling node is selected. However, if the sibling node is expandable, we need to check if it is expanded. If the sibling node is **not** expanded, we render an active guideline. Otherwise, we do not render an active guideline.

Then, the `nodePtr` points to the parent of the parent node and repeats the steps above for other indents of the tree node that is currently rendered.